test/addons/buffer-free-callback/binding.cc

```diff
@@ -0,0 +1,38 @@
+#include <node.h>
+#include <node_buffer.h>
+#include <util.h>
+#include <v8.h>
+
+static bool alive = false;
+static char buf[1024];
+
+static void FreeCallback(char* data, void* hint) {
+  alive = false;
+}
+
+void Alloc(const v8::FunctionCallbackInfo<v8::Value>& args) {
+  v8::Isolate* isolate = args.GetIsolate();
+  v8::HandleScope scope(isolate);
+  alive = true;
+  args.GetReturnValue().Set(node::Buffer::New(
+        isolate,
+        buf,
+        sizeof(buf),
+        FreeCallback,
+        nullptr).ToLocalChecked());
+}
+
+void Check(const v8::FunctionCallbackInfo<v8::Value>& args) {
+  v8::Isolate* isolate = args.GetIsolate();
+  v8::HandleScope scope(isolate);
+  isolate->RequestGarbageCollectionForTesting(
+      v8::Isolate::kFullGarbageCollection);
```

Comment:
Out of curiosity, is it viable to use a technique similar to https://groups.google.com/d/msg/v8-users/JmhzmswiqvM/IC6UtV5FEC4J so that `--expose-gc` doesn't have to be passed?